light-client-verifier/src/verifier.rs

```diff
@@ -490,4 +490,100 @@ mod tests {
             .to_string()
             .contains("invalid validator address"),);
     }
+
+    #[test]
+    #[cfg(feature = "rust-crypto")]
+    fn test_successful_verify_malicious_update_header_2() {
+        // This test is almost same as the previous one.
+        // The only difference is order of the validator set iteration.
+        // In this case, the verifier will correctly detect invalid signature.
```

Comment:
What's the other case?

Reply:
`test_successful_verify_malicious_update_header_1` is the other case.

Basically, a malicious validator could only reuse its vote if only it came earlier than the targeted validator in this following loop (`test_successful_verify_malicious_update_header_1`).

https://github.com/informalsystems/tendermint-rs/blob/71c946084eceed3a2da91b78921d0345b46401e5/light-client-verifier/src/operations/voting_power.rs#L419-L420

If the targeted validator came earlier in this loop, the verifier will see `vote.verified` is false and (in)validate the signature (`test_successful_verify_malicious_update_header_2`).

Reply:
check 73dec22fa3eb3847861aa9fc511530ed1fc1cbf0 for the changes I made to the existing test to trigger the invalid signature error.